app/questionnaire/questionnaire_schema.py

```diff
@@ -215,20 +216,52 @@ def get_post_submission(self) -> ImmutableDict:
         schema: ImmutableDict = self.json.get("post_submission", ImmutableDict({}))
         return schema
 
+    def _is_list_name_in_rule(
+        self, when_rules: Union[dict, Sequence], list_name: str
+    ) -> bool:
+        rules: Union[dict, Sequence]
+        if isinstance(when_rules, dict) and any(
+            operator in when_rules for operator in OPERATION_MAPPING
+        ):
+            rules = self.get_operand(when_rules)
+        else:
+            rules = when_rules
+
+        if isinstance(rules, Sequence):
```

Comment:
When would it be a dict here?

Reply:
It will be always `sequence` , if it is a `dict` then we work out( before) if it has `operator` in  then we get operator values which could be a rule or another operator with rules as below

```
                 "when": { 
                            ">": [
                                        {
                                            "count": [
                                                {
                                                    "source": "list",
                                                    "identifier": "people"
                                                }
                                            ]
                                        },
                                        1
                                    ]
                                }
                            },
```